core/src/test/java/org/frankframework/senders/ReconnectSenderWrapperTest.java

```diff
@@ -32,39 +36,57 @@ public void basic() throws Exception {
 		assertEquals(input, result);
 	}
 
+	@Test
+	void testIfSenderIsClosed() throws Exception {
+		// Arrange
+		ISender senderMock = Mockito.mock(ISender.class);
+		when(senderMock.sendMessage(Mockito.any(Message.class), Mockito.any(PipeLineSession.class))).thenReturn(new SenderResult(Message.nullMessage()));
+		sender.setSender(senderMock);
+		sender.configure();
+		sender.open();
+
+		// Act
+		sender.sendMessageOrThrow(Message.nullMessage(), session).asString();
+
+		// Assert
+		verify(senderMock, Mockito.times(2)).open();
+		verify(senderMock, Mockito.times(1)).close();
```

Comment:
Perhaps it's an idea to close the session, and verify that close is called once more.

Reply:
Yes indeed, I've improved it with your idea.